src/components/number/number.test.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Number from ".";
+import Logger from "../../__internal__/utils/logger";
+
+jest.mock("../../__internal__/utils/logger");
+
+const ControlledNumber = () => {
+  const [state, setState] = useState("");
+  const setValue = ({ target }: React.ChangeEvent<HTMLInputElement>) => {
+    setState(target.value);
+  };
+  return <Number value={state} onChange={setValue} />;
+};
+
+test("renders a textbox with provided label", () => {
+  render(<Number label="foo" />);
+
+  expect(screen.getByRole("textbox", { name: "foo" })).toBeVisible();
+});
+
+test("renders a textbox with provided value", () => {
+  render(<Number value="123" />);
+
+  expect(screen.getByRole("textbox")).toHaveValue("123");
+});
+
+test("calls onChange when valid characters are provided", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(<Number onChange={onChange} />);
+
+  await user.tab();
+  await user.keyboard("123");
```

Comment:
**suggestion** (non-blocking): user-event provides the `type` utility specifically for typing into input elements, which I feel is more appropriate to use for this component that `user.keyboard` (and also removes the need to manually tab into or otherwise focus the input) https://testing-library.com/docs/user-event/utility#type

Same applies to all uses of `user.keyboard` in here where we're just entering text into the input